curriculum/challenges/_meta/functional-programming/meta.json

```diff
@@ -68,6 +68,10 @@
       "587d7da9367417b2b2512b68",
       "Use the reduce Method to Analyze Data"
     ],
+    [
+      "587d7b88367417b2b2512b45",
+      "Use Higher-Order Functions map, filter, or reduce to Solve a Complex Problem"
```

Comment:
```suggestion
      "Use Higher-Order Functions to Solve a Complex Problem"
```

Reply:
The reason I chose that title is because even though `forEach` is a higher order function, we don't want them using it.  The purpose of the challenge is for the user to use map, filter, and/or reduce, but I suppose as long as the instructions state what is expected, I can change it.

Reply:
I guess we should change it then. Maybe @ojeytonwilliams and @moT01 can provide more on it

Reply:
This isn't feedback on this PR, but a quick two cents about what I think - The order of the javascript section never seemed right to me. Why would ES6 be the second thing taught? I know, we want people to get right into the newest stuff. But, you would think it would be something like `basic javascript`, `basic data structures`, `functional programming`, then maybe `es6` - This section just doesn't feel like it follows a natural progression. If that were the order, this issue would never have come up. Regex is the third thing taught in this section? why? --- rant
That's where I think the heart of this issue lies.

I'll try and take a closer look at this PR at some point today to see give some better feedback.

Reply:
We definitely should have a discussion somewhere concerning the overall section order (if an issue does not already exist for this).

Reply:
Both titles seem fine, honestly.  I'd be perfectly happy if either ended up being used.

This new spot for the challenge does seem a lot better, though.

Reply:
@Manish-Giri Could you take a look at this PR (the title in particular) and see what you think?

Reply:
Sorry for being late to the party!

I like the idea of being explicit in the title - 
> Use Higher-Order Functions Map, Filter, or Reduce to Solve a Complex Problem

As there are quite a few higher order functions available, this verbiage in the title itself feels more direct in terms of what is expected of the camper, even though the challenge instructions also specify what is required.

Another point is - `map()`, `reduce()` and `filter()` are some of the more commonly used functions throughout ES6 and functional programming (in general), so it's probably good to have a challenge with these functions in the title. If I were looking to solve just a few select challenges in the curriculum (to brush up on ES6 concepts), that title would be beneficial.